applications/GeoMechanicsApplication/tests/cpp_tests/test_transport_equation_utilities.cpp

```diff
@@ -134,4 +141,67 @@ KRATOS_TEST_CASE_IN_SUITE(CalculateBiotCoefficients_GivesInfResults_ForZeroBulkM
                                    [](const double value) { return std::isinf(value); }))
 }
 
+KRATOS_TEST_CASE_IN_SUITE(EachFluidPressureIsTheInnerProductOfShapeFunctionsAndPressure, KratosGeoMechanicsFastSuite)
+{
+    auto shape_function_values = Matrix{3, 3, 0.0};
+    // clang-format off
+    shape_function_values <<= 0.8, 0.2, 0.3,
+                              0.1, 0.7, 0.4,
+                              0.2, 0.5, 0.6;
+    // clang-format on
+
+    auto pore_water_pressures = Vector(3);
+    pore_water_pressures <<= 2.0, 3.0, 4.0;
+
+    const auto expected_fluid_pressures = std::vector<double>{3.4, 3.9, 4.3};
+    KRATOS_EXPECT_VECTOR_NEAR(GeoTransportEquationUtilities::CalculateFluidPressures(
+                                  shape_function_values, pore_water_pressures),
+                              expected_fluid_pressures, tolerance)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(PermeabilityUpdateFactorEqualsOneWhenChangeInverseFactorIsNotGiven, KratosGeoMechanicsFastSuite)
+{
+    const auto unused_strain_vectors = std::vector<Vector>(3, Vector{});
+    auto       properties            = Properties{};
+
+    const auto expected_factors = std::vector<double>(unused_strain_vectors.size(), 1.0);
+    KRATOS_EXPECT_VECTOR_NEAR(GeoTransportEquationUtilities::CalculatePermeabilityUpdateFactors(
+                                  unused_strain_vectors, properties),
+                              expected_factors, tolerance)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(PermeabilityUpdateFactorEqualsOneWhenChangeInverseFactorIsNonPositive, KratosGeoMechanicsFastSuite)
+{
+    const auto unused_strain_vectors = std::vector<Vector>(3, Vector{});
+    auto       properties            = Properties{};
+    properties[PERMEABILITY_CHANGE_INVERSE_FACTOR] = -1.0;
+
+    const auto expected_factors = std::vector<double>(unused_strain_vectors.size(), 1.0);
+    KRATOS_EXPECT_VECTOR_NEAR(GeoTransportEquationUtilities::CalculatePermeabilityUpdateFactors(
+                                  unused_strain_vectors, properties),
+                              expected_factors, tolerance)
+
+    properties[PERMEABILITY_CHANGE_INVERSE_FACTOR] = 0.0;
+
+    KRATOS_EXPECT_VECTOR_NEAR(GeoTransportEquationUtilities::CalculatePermeabilityUpdateFactors(
+                                  unused_strain_vectors, properties),
+                              expected_factors, tolerance)
+}
```

Comment:
Here this might be a similar question for a next issue: where will we check the PERMEABILITY_CHANGE_INVERSE_FACTOR and is this expected behavior for the function

Reply:
I honestly don't know... Perhaps this is something that we may need to discuss with the rest of the team?

Reply:
I created an issue for checking some other user defined material parameters, so we include it there 👍